ee/tpmrunner/tpmrunner.go

```diff
@@ -0,0 +1,241 @@
+package tpmrunner
+
+import (
+	"context"
+	"crypto"
+	"errors"
+	"fmt"
+	"io"
+	"log/slog"
+	"time"
+
+	"github.com/kolide/krypto/pkg/tpm"
+	"github.com/kolide/launcher/ee/agent/types"
+	"github.com/kolide/launcher/pkg/traces"
+)
+
+type tpmRunner struct {
+	signer        crypto.Signer
+	signerCreator tpmSignerCreator
+	store         types.GetterSetterDeleter
+	slogger       *slog.Logger
+	interrupt     chan struct{}
+	interrupted   bool
+}
+
+// tpmSignerCreator is an interface for creating and loading TPM signers
+// useful for mocking in tests
+type tpmSignerCreator interface {
+	CreateKey(opts ...tpm.TpmSignerOption) (private []byte, public []byte, err error)
+	New(private, public []byte) (crypto.Signer, error)
+}
+
+// defaultTpmSignerCreator is the default implementation of tpmSignerCreator
+// using the tpm package
+type defaultTpmSignerCreator struct{}
+
+// CreateKey creates a new TPM key
+func (d defaultTpmSignerCreator) CreateKey(opts ...tpm.TpmSignerOption) (private []byte, public []byte, err error) {
+	return tpm.CreateKey()
+}
+
+// New creates a new TPM signer
+func (d defaultTpmSignerCreator) New(private, public []byte) (crypto.Signer, error) {
+	return tpm.New(private, public)
+}
+
+// tpmRunnerOption is a functional option for tpmRunner
+// useful for setting dependencies in tests
+type tpmRunnerOption func(*tpmRunner)
+
+func New(ctx context.Context, slogger *slog.Logger, store types.GetterSetterDeleter, opts ...tpmRunnerOption) (*tpmRunner, error) {
+	_, span := traces.StartSpan(ctx)
+	defer span.End()
+
+	_, _, err := fetchKeyData(store)
+	if err != nil {
+		return nil, err
+	}
```

Comment:
Why do we have to do this here? I'm curious because it doesn't seem to have any side effects. (Also, if we do have to do this but not _here_ necessarily, maybe we could move it to the top of `Execute` to not delay rungroup startup.)

Reply:
It just verifies that the provided store is valid. I think your comment [here](https://github.com/kolide/launcher/pull/1977#discussion_r1873440394) applies to this as well. I think you're right, we should move these ops to execute, there is very little chance that the store will be invalid and if it ever was for some reason, I'm willing to delay that discovery to the execute loop.

Reply:
moved to top of execute